### ISSUE TITLE
patternProperties does not work

### ISSUE THREAD

**Issue by piotrkochan** [justinrainbow/json-schema#584]
schema: 

```
{
  "$schema": "http://json-schema.org/draft-06/schema#",
  "$id": "http://json-schema.org/draft-06/schema#",
  "type": "object",
  "patternProperties": {
    "^[A-Z]{2}+$": {
      "type": "array",
      "items": {
        "type": "string",
        "maxLength": 2,
        "minLength": 2
      }
    }
  }
}
```

Object:
```
{"sdfsdf":["aa","bb"]}
```
false positive:

The supplied JSON validates against the schema.

**Comment by piotrkochan** [comment]
Tried for same object:

```"patternProperties": {
    "^[0-9]+": {```

Same output: `The supplied JSON validates against the schema.`

**Comment by piotrkochan** [comment]
```
"patternProperties": {
    "sdfsdfsdf": {
```
Oh come on `The supplied JSON validates against the schema.`

**Comment by piotrkochan** [comment]
https://github.com/justinrainbow/json-schema/pull/585

**Comment by erayd** [comment]
Your sample data is supposed to be valid, as it does not match any rule which restricts it. This behavior is correct.

**Comment by piotrkochan** [comment]
Yeah, pattern is totally ignored. Great feature.

**Comment by erayd** [comment]
It isn't ignored. There is just no property which matches it.

The point of `patternProperties` is to apply a subschema to any property with a name that matches the pattern. Any property which does *not* match the pattern is irrelevant, and is not constrained by the subschema. Unless there is another constraint elsewhere that is violated by that property, it will be considered valid - the spec mandates that any value is valid by default.

**Comment by shmax** [comment]
Great explanation, thank you. 👍
